src/nagini_contracts/lock.py

```diff
@@ -37,17 +47,35 @@ def __init__(self,
                          WaitLevel() < Level(below)))
         Requires(Implies(above is not None and below is not None,
                          Level(above) < Level(below)))
```

Comment:
I think the idea is that the newly created lock's level is above that of 'above' and below that of 'below'. I'll add a sentence to the docstring that explains it.

Reply:
Ok, I would suggest renaming above->low_level, below->high_level or something similar, so that you have a clear 'Level(low_level) < Level(high_level)'. But that's just a suggestion.